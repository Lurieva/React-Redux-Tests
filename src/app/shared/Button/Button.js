import React from 'react';
import PropTypes from 'prop-types';

const Button = ({ className, disabled, onClick, name }) => (
    <button className={className}
            disabled={disabled}
            onClick={onClick}>
        {name}
    </button>
);

Button.propTypes = {  
    name: PropTypes.string.isRequired,
    onClick: PropTypes.func.isRequired,
    disabled: PropTypes.string,
    className: PropTypes.string
};

export default Button;