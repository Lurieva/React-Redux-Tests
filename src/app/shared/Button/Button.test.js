import React from 'react';
import { shallow } from 'enzyme';
import Button from './Button';

describe('Button', () => {
    const mockFn = jest.fn();

    it('should be defined', () => {
        expect(Button).toBeDefined();
    });

    it('should render correctly', () => {
        const tree = shallow(
          <Button className='classNameMock' disabled='disabled' name='BtnName' onClick={mockFn} />
        );

        expect(tree).toMatchSnapshot();
    });

    it('should call mock function when button click', () => {
        const tree = shallow(
            <Button className='classNameMock' disabled='disabled' name='BtnName' onClick={mockFn} />
        );

        tree.simulate('click');
        expect(mockFn).toHaveBeenCalled();
    });
});