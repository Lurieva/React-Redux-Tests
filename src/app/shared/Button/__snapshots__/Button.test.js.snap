// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    className="classNameMock"
    disabled="disabled"
    name="BtnName"
    onClick={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "BtnName",
      "className": "classNameMock",
      "disabled": "disabled",
      "onClick": [MockFunction],
    },
    "ref": null,
    "rendered": "BtnName",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "BtnName",
        "className": "classNameMock",
        "disabled": "disabled",
        "onClick": [MockFunction],
      },
      "ref": null,
      "rendered": "BtnName",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
